sources/packages/backend/libs/src-sql/sfas-restrictions/Bulk-insert-restrictions.sql

```diff
@@ -0,0 +1,118 @@
+/* 
+ * After all the SFAS restrictions are imported, 
+ * the below query inserts one Legacy restriction entry 
+ * per student if they have one or more restrictions 
+ * imported from the sfas system that are not present 
+ * in the sims restrictions.
+ */
+INSERT INTO
+  sims.student_restrictions (student_id, restriction_id)
+SELECT
+  students.student_id,
+  (
+    SELECT
+      restrictions.id
+    FROM
+      sims.restrictions restrictions
+    WHERE
+      restrictions.restriction_code = 'LGCY'
+  ) AS restriction_id
+FROM
+  (
+    (
+      SELECT
+        student_restrictions.student_id AS student_id
+      FROM
+        sims.student_restrictions student_restrictions
+      WHERE
+        student_restrictions.restriction_id = (
+          SELECT
+            id
+          FROM
+            sims.restrictions
+          WHERE
+            restriction_code = 'LGCY'
+        )
+        AND student_restrictions.is_active = true
```

Comment:
I would assume that this part of the query is trying to avoid inserting a new `LGCY`  if one is already present, right? Would it not simplify the query to add an exists condition to the final where maybe?